Add mcxn236 smartdma support

NeilChen93 opened the pull request:
Add support for using the SMARTDMA engine on the FRDM-MCXN236 board with the video capture sample.

NeilChen93 committed:
dts: arm/nxp: Add smartdma nodes to NXP MCXN23x dtsi file

Add smartdma nodes to NXP MCXN23x dtsi file

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
boards: nxp: frdm_mcxn236: Support i3c for NXP frdm_mcxn236 board

Support smartdma and flexcomm0 for NXP frdm_mcxn236 board.
The default console pins are duplicated with camera header pins, so
add flexcomm0_lpuart0 for smartdma_camera serial output.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

ngphibang commented on samples/drivers/video/capture/boards/frdm_mcxn236.overlay:
```diff
@@ -0,0 +1,21 @@
+ */
+
+/ {
+	chosen {
+		zephyr,uart-mcumgr = &flexcomm0_lpuart0;
+		zephyr,console = &flexcomm0_lpuart0;
+		zephyr,shell-uart = &flexcomm0_lpuart0;
+	};
```
It seems these chosen node are not video-specific. Could you explain why we don't set it elsewhere (e.g, board's dts) but need to set it here in the video capture sample overlay ?

NeilChen93 replied:
Hi @ngphibang , the default console pins(P1_8,P1_9) are duplicated with camera header. So I chose another flexcomm(flexcomm0_lpuart0) for smartdma_camera serial output.
![image](https://github.com/user-attachments/assets/05d2a6b7-258a-43cc-b5b4-82cc58685c5e)

ngphibang commented on samples/drivers/video/capture/boards/frdm_mcxn236.conf:
```diff
@@ -0,0 +1,2 @@
+CONFIG_VIDEO_BUFFER_POOL_SZ_MAX=40000
+CONFIG_VIDEO_BUFFER_POOL_NUM_MAX=2
```
`CONFIG_VIDEO_BUFFER_POOL_NUM_MAX` is set to 2 by default so this is not needed.

hakehuang commented:
@NeilChen93 the sample fails on frdm_mcxn947, due to hal upgrade sdma, so it is fixed now?

ngphibang commented:
@hakehuang It is fixed by https://github.com/zephyrproject-rtos/zephyr/pull/91760#event-18229132453 (sorry not to inform you as I think you were aware of it). The problem was finally found due to ov7670 init sequence (not sdma)

hakehuang commented:
> @hakehuang It is fixed by [#91760 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/91760#event-18229132453) (sorry not to inform you as I think you were aware of it). The problem was finally found due to ov7670 init sequence (not sdma)

@ngphibang, @NeilChen93  can you help to add `- video` to those boards frdm_mcxn947_cpu0 and frdm_mcxn236

NeilChen93 committed:
samples: video: Add support for smartdma video on FRDM_MCXN236

Add support for using the SMARTDMA engine on the FRDM-MCXN236 board with
the video capture sample.
Replace console pins, because default console pins(P1_8, P1_9) are
duplicated with camera header pins.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
boards: nxp: frdm_mcxn947: Add video support in frdm_mcxn947 board yaml

Add video support in frdm_mcxn947 board yaml

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

ngphibang reviewed (dismissed):
LGTM

hakehuang reviewed (commented):
add `- video`  to <platform.yaml> support list

josuah commented on boards/nxp/frdm_mcxn236/frdm_mcxn236.dtsi:
```diff
@@ -185,3 +192,39 @@ zephyr_mipi_dbi_parallel: &flexio0_lcd {
+
+dvp_20pin_i2c: &flexcomm2_lpi2c2 {};
+
+dvp_20pin_interface: &video_sdma {};
```
This line should have been enough for allowing the `chosen { zephyr,camera = <...>; }` to take effect through the shield:
https://github.com/zephyrproject-rtos/zephyr/blob/6c68584046046e42e5c98efecb12e2ea6bad2d27/boards/shields/dvp_20pin_ov7670/dvp_20pin_ov7670.overlay#L9-L11

Not sure yet why there is the [CI error](https://github.com/zephyrproject-rtos/zephyr/actions/runs/15869626626/job/44743372926?pr=92058#step:13:641)...

```
/__w/zephyr/zephyr/include/zephyr/device.h:96:41: error: '__device_dts_ord_DT_CHOSEN_zephyr_camera_ORD' undeclared (first use in this function)
   96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                         ^~~~~~~~~
```

ngphibang replied:
Maybe because of not use the full board name `frdm_mcxn236/mcxn236` ?

josuah reviewed (commented):
In case the error for MCXN236 is the same as the error for MCXN947:
- https://github.com/zephyrproject-rtos/zephyr/issues/92299

If so, then this PR is probably not causing this error.

josuah reviewed (dismissed):
This works with UVC (`git rebase main` of this branch).
![mpv-shot0001](https://github.com/user-attachments/assets/29bd7c48-d516-4e3a-8eff-eaf1b4882f92)

ngphibang commented on samples/drivers/video/capture/sample.yaml:
```diff
@@ -30,6 +31,7 @@ tests:
       - mimxrt1170_evk/mimxrt1176/cm7
       - mimxrt1170_evk@B/mimxrt1176/cm7
       - frdm_mcxn947/mcxn947/cpu0
+      - frdm_mcxn236
```
Should be full name `frdm_mcxn236/mcxn236` ?

ngphibang commented on samples/drivers/video/capture/sample.yaml:
```diff
@@ -11,6 +11,7 @@ tests:
       - platform:mimxrt1170_evk/mimxrt1176/cm7:SHIELD="nxp_btb44_ov5640;rk055hdmipi4ma0"
       - platform:mimxrt1170_evk@B/mimxrt1176/cm7:SHIELD="nxp_btb44_ov5640;rk055hdmipi4ma0"
       - platform:frdm_mcxn947/mcxn947/cpu0:SHIELD="dvp_20pin_ov7670;lcd_par_s035_8080"
+      - platform:frdm_mcxn236:SHIELD="dvp_20pin_ov7670;lcd_par_s035_8080"
```
Should be full name `frdm_mcxn236/mcxn236` ?

hakehuang commented:
> I will get another camera to try, but it looks not like camera issue

I change to another camera which has the same problem.

hakehuang reviewed (approved):
there is a failure in samples/driver/video/capture which reported in #92299
but not related to this PR.

josuah reviewed (approved):
LGTM thank you for the changes!
I am adding a DNM flag so that it does not accidentally arrive in the merge-list before NXP and others could decide if it is ok to add it now that we are in feature-freeze (4.2-rc1 released)

Feel free to remove the DNM flag!

ngphibang reviewed (approved):
LGTM. The failure with mcxn947 is not related to this PR (`CONFIG_TEST` is an additional test that helps to automatically detects issues without human check, it is not a CI test to pass ...)

danieldegrasse commented:
Moving to 4.3 milestone as this is a new feature
